src/components/form.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ChangeEvent, DetailedHTMLProps, FormHTMLAttributes, PropsWithChildren, ReactElement, createContext, useContext } from "react";
+import { useForm } from "../utils/use-form";
+
+export interface FormProps extends PropsWithChildren<DetailedHTMLProps<FormHTMLAttributes<HTMLFormElement>, HTMLFormElement>>{
+    handleSubmit : (data: Record<string,string>) => void;
+    initialState : Record<string,string>;
+}
+
+interface FormContextType{
+    handleInputChange: (event: ChangeEvent<HTMLInputElement>) => void;
+    formState: Record<string, string>;
+}
+
+export const FormContext = createContext<FormContextType>({
+    handleInputChange: (event: ChangeEvent<HTMLInputElement>) => {},
+    formState: {}
+});
+
+export default function Form({initialState, handleSubmit, children, ...props}: FormProps): ReactElement{
+    const {formState, handleInputChange, handleSubmit: handleSubmitForm} = useForm(initialState);
+
+    return(
+        <form onSubmit={(event) => {handleSubmitForm(event, handleSubmit)}} {...props}>
+            <FormContext.Provider value={{handleInputChange, formState}}>
+                {children}
+            </FormContext.Provider>
+        </form>
+    )
+}
```

Comment:
En général sur se genre d'outil j'ai bien faire un hook personalisé genre useForm() et useField() ou il y aura la logique d'utilisation directement avec le context, pour pas devoir importer le context à droite à gauche

```suggestion
}
const useForm = ()=>{
    return useContext(...)
}
const useField = (name)=>{
    const {form, setForm} =  useContext(...)
    
    const setField = (value)=>{
        setForm({...form,[name]:value});
    }
    return [form[name], setField];
}

```

Tu peux aussi faire une synchronisation des erreurs si jamais :)